src/Services/CheckoutService.php

```diff
@@ -1,251 +0,0 @@
-<?php
-
-declare(strict_types=1);
-
-namespace Tipoff\Checkout\Services;
-
-use Illuminate\Database\Eloquent\Model;
-use Tipoff\Checkout\Models\Cart;
-
-class CheckoutService
-{
-    /* Tax & fee application methods */
-    const APPLICATION_ORDER = 'order';
-    const APPLICATION_EACH = 'each';
-    const APPLICATION_PRODUCT = 'product';
-    const APPLICATION_BOOKING = 'booking';
-    const APPLICATION_PARTICIPANT = 'participant';
-
-    /**
-     * Check if discount/voucher code is valid.
-     *
-     * @param string $code
-     * @return mixed
-     * @psalm-suppress UndefinedMethod
-     */
-    public function findDeductionByCode($code)
```

Comment:
Moved to Cart model as private helper.   If the resulting deduction is needed external to this package, it could be promoted to a CartInterface method.